### ISSUE TITLE
How to get calendarYear strata from the targerCohort after intersecting

### ISSUE THREAD

**Issue by cebarboza** [darwin-eu-dev/PatientProfiles#415]
Hey guys, I have a question on how to do an intersect. I already created a calendarYear into my condition cohort (hepatitis).

```R
# Condition
cdm[[cohortNameCondition]] <- cdm[[cohortNameCondition]] %>%
  PatientProfiles::addAge(ageGroup = list(c(1, 17),
                                          c(18, 44),
                                          c(45, 64),
                                          c(65, 150))) %>%
  PatientProfiles::addSex() %>%
  dplyr::mutate(calendarYear = !!CDMConnector::datepart("cohort_start_date")) %>%
  CDMConnector::computeQuery()
 
```
```
  cohort_definition_id subject_id cohort_start_date cohort_end_date   age age_group sex    calendarYear
                 <int>    <int64> <date>            <date>          <dbl> <chr>     <chr>         <dbl>
1                    1          3 2015-06-07        2017-11-01         50 45 to 64  Female         2015
2                    1          4 2015-05-01        2020-06-05          5 1 to 17   Female         2015
3                    1          5 2013-05-02        2015-04-01         77 65 to 150 Female         2013
4                    1          7 2019-02-05        2022-12-30         31 18 to 44  Female         2019
5                    1          9 2019-02-01        2022-12-01         29 18 to 44  Male           2019
```
```R
# Target cohort

> cdm$hepatitisb_treatment
# Source:   table<hepatitisb_treatment> [2 x 4]
# Database: postgres  [postgres@localhost:5432/testcdm]
  cohort_definition_id subject_id cohort_start_date cohort_end_date
                 <int>    <int64> <date>            <date>         
1                    1          4 2015-06-01        2020-06-05     
2                    1          8 2015-08-05        2022-06-01  

```

But when I do the intersection, I get the strata for calendarYear from my Condition cohort, not from the Target cohort. 

```R
  summaryTreatment <- cdm[[cohortNameCondition]] %>%
    summariseCharacteristics(
      strata = list("age_group",
                    "sex",
                    "calendarYear"),
      tableIntersect = list(),
      cohortIntersect = list(
        "Treatment" = list(
          targetCohortTable = cohortNameTreatment,
          targetEndDate = NULL,
          value = "flag",
          window = list(c(0, Inf))
        )),
      minCellCount = minCellCount
    )
```
```
# A tibble: 3 × 11
  cdm_name result_type             group_name  group_level strata_name  strata_level variable        variable_level variable_type estimate_type estimate
  <chr>    <chr>                   <chr>       <chr>       <chr>        <chr>        <chr>           <chr>          <chr>         <chr>         <chr>   
1 testcdm  Summary characteristics cohort_name Hepatitis b calendarYear 2013         Number subjects NA             numeric       count         1       
2 testcdm  Summary characteristics cohort_name Hepatitis b calendarYear 2015         Number subjects NA             numeric       count         2       
3 testcdm  Summary characteristics cohort_name Hepatitis b calendarYear 2019         Number subjects NA             numeric       count         2  
```

So what I did was to insert into my Condition cohort the calendar years from the intersect with the target cohort. So when I do the summariseCharacteristics intersect, it gets the calendarYear strata from the target condition. 

```R
  cdm[[cohortNameCondition]] <- cdm[[cohortNameCondition]] %>%
    addCohortIntersect(targetCohortTable = cohortNameTreatment,
                       targetEndDate = NULL,
                       flag = FALSE,
                       count = FALSE,
                       date = TRUE,
                       days = FALSE,
                       nameStyle = "calendarYearInterferons")


  cdm[[cohortNameCondition]] <- cdm[[cohortNameCondition]] %>%
    dplyr::mutate(calendarYear = !!CDMConnector::datepart("calendaryearinterferons"))
```
```
  cohort_definition_id subject_id cohort_start_date cohort_end_date   age age_group sex    calendarYear calendaryearinterferons
                 <int>    <int64> <date>            <date>          <dbl> <chr>     <chr>         <dbl> <date>                 
1                    1          4 2015-05-01        2020-06-05          5 1 to 17   Female         2015 2015-06-01             
2                    1          7 2019-02-05        2022-12-30         31 18 to 44  Female           NA NA                     
3                    1          3 2015-06-07        2017-11-01         50 45 to 64  Female           NA NA                     
4                    1          9 2019-02-01        2022-12-01         29 18 to 44  Male             NA NA                     
5                    1          5 2013-05-02        2015-04-01         77 65 to 150 Female           NA NA    
```
Summary
```
# A tibble: 2 × 11
  cdm_name result_type             group_name  group_level strata_name  strata_level variable        variable_level variable_type estimate_type estimate
  <chr>    <chr>                   <chr>       <chr>       <chr>        <chr>        <chr>           <chr>          <chr>         <chr>         <chr>   
1 testcdm  Summary characteristics cohort_name Hepatitis b calendarYear 2015         Number subjects NA             numeric       count         1       
2 testcdm  Summary characteristics cohort_name Hepatitis b calendarYear Na           Number subjects NA             numeric       count         4     
```

Is there a way to do this in a more simple way? The thing is that we are calculating for every ingrediente, and we would like the strata for calendarYear for the target cohort.

**Comment by cebarboza** [comment]
When calculating from a cohort with multiple cohort_ids, I am trying the same approach. But end up with a table with many columns from every intersect. 

```R
    cohortDrugList <- CodelistGenerator::codesFromConceptSet(system.file("cohorts",
                                                                                                      conditionType,
                                                                                                      objectiveType,
                                                                                                      "drugs",
                                                                                                      package = "P2C1010ChronicHepatitis"),
                                                                                                      cdm = cdm,
                                                                                                      withConceptDetails = FALSE)

  names(cohortDrugList) <- gsub(" ", "_", tolower(names(cohortDrugList)))
  names(cohortDrugList) <- gsub("2", "x", names(cohortDrugList))

  drugsTableName <- glue::glue("{objectiveType}_drugs")
  
  cdm <- CDMConnector::generateConceptCohortSet(cdm = cdm,
                                                name = drugsTableName,
                                                conceptSet = cohortDrugList,
                                                limit = "all",
                                                end = "observation_period_end_date",
                                                overwrite = TRUE)
                                                
  # Date Intersect
  
  cdm[[cohortNameCondition]] <- cdm[[cohortNameCondition]] %>%
    addCohortIntersectDate(targetCohortTable = drugsTableName,
                           nameStyle = "{cohort_name}_years")

```

```
# Source:   table<dbplyr_046> [5 x 22]
# Database: postgres  [postgres@localhost:5432/testcdm]
  cohort_definition_id subject_id cohort_start_date cohort_end_date   age age_group sex    calendarYear calendaryearinterferons pegxa_years lamivudine_years
                 <int>    <int64> <date>            <date>          <dbl> <chr>     <chr>         <dbl> <date>                  <date>      <date>          
1                    1          5 2013-05-02        2015-04-01         77 65 to 150 Female           NA NA                      NA          NA              
2                    1          4 2015-05-01        2020-06-05          5 1 to 17   Female         2015 2015-06-01              NA          2015-06-01      
3                    1          9 2019-02-01        2022-12-01         29 18 to 44  Male             NA NA                      NA          2020-09-01      
4                    1          7 2019-02-05        2022-12-30         31 18 to 44  Female           NA NA                      NA          2019-03-05      
5                    1          3 2015-06-07        2017-11-01         50 45 to 64  Female           NA NA                      NA          NA              
# ℹ 11 more variables: tenofovirx_years <dbl>, cep_years <dbl>, tenofovir_years <dbl>, pegx_years <dbl>, bulevirtide_years <dbl>, pegxb_years <dbl>,
#   interferonxb_years <dbl>, telbivudine_years <dbl>, entecavir_years <dbl>, interferon_years <dbl>, adefovir_years <dbl>
```

**Comment by catalamarti** [comment]
should be fixed with #419
